### PR TITLE
Fix : run the  embedded elasticsearch only for the unit tests that need it

### PR DESCRIPTION
**Pull request by SlimaneAmar**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ x] The commit message follows our guidelines

**What is the current behavior?** *(You can also link to an open issue here)*
The  embedded elasticsearch run for all unit tests

**What is the new behavior (if this is a feature change)?**
The  embedded elasticsearch run for only unit tests  that need it

### CODE REVIEW

File: src/test/java/com/powsybl/network/conversion/server/EmbeddedElasticsearch.java [hunk 1]
```diff
@@ -24,9 +26,12 @@
 
     private static ElasticsearchContainer elasticsearchContainer;
 
+    @Value("${spring.data.elasticsearch.enabled:false}")
+    boolean elasticsearchEnabled;
+
     @PostConstruct
     public void postConstruct() {
-        if (elasticsearchContainer != null) {
+        if (elasticsearchContainer != null || !elasticsearchEnabled) {
```
Comment (niconoir) [inline src/test/java/com/powsybl/network/conversion/server/EmbeddedElasticsearch.java:34]: What if `!elasticsearchEnabled` **AND** `elasticsearchContainer != null` ? Maybe it should be ensured the elasticsearchContainer is stopped in this case ?

File: src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java [hunk 2]
```diff
@@ -64,8 +63,7 @@
  */
 
 @RunWith(SpringRunner.class)
-@WebMvcTest(value = NetworkConversionController.class, properties = {"spring.data.elasticsearch.enabled=true"})
-@ContextConfiguration(classes = {NetworkConversionApplication.class, NetworkConversionService.class, EmbeddedElasticsearch.class})
+@WebMvcTest(value = NetworkConversionController.class)
```
Comment (niconoir) [inline src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java:66]: Why did you completely remove line `@ContextConfiguration(classes = {NetworkConversionApplication.class, NetworkConversionService.class, EmbeddedElasticsearch.class})` ?

File: src/test/java/com/powsybl/network/conversion/server/EmbeddedElasticsearch.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/test/java/com/powsybl/network/conversion/server/EmbeddedElasticsearch.java:34]: Only one embedded elasticsearch is started for all test classes (static attribute)
And all EmbeddedElasticsearch beans are destroyed in the end of all test classes

File: src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java:66]: No need to indicate the context configuration (just a cleaning)

**Review by niconoir — APPROVED** [review]

**Pull request merged**